core/kazoo_documents/src/kzd_box_message.erl

```diff
@@ -74,21 +74,23 @@
 %%--------------------------------------------------------------------
 -spec new(ne_binary(), kz_proplist()) -> doc().
 new(AccountId, Props) ->
-    {Year, Month, _} = erlang:date(),
+    UtcSeconds = props:get_value(<<"Message-Timestamp">>, Props, kz_util:current_tstamp()),
+    Timestamp  = props:get_value(<<"Document-Timestamp">>, Props, UtcSeconds),
+    {Year, Month, _} = kz_util:to_date(Timestamp),
```

Comment:
Note the existence of https://github.com/2600hz/kazoo/blob/master/core/kazoo/src/kz_util.erl#L243.